great_expectations/expectations/core/expect_column_values_to_be_of_type.py

```diff
@@ -566,7 +566,7 @@ def _validate(
         actual_column_type = [
             type_dict["type"]
             for type_dict in actual_column_types_list
-            if type_dict["name"] == column_name
+            if type_dict["name"].casefold() == column_name.casefold()
```

Comment:
Thanks for contributing. I agree that an IndexError is onerous for a user to debug and we should make this easier. I don't this this is a viable solution though because a table can have column names where casing matters. For example, a postgres database may have a table, `MyTable`, with quoted columns of different types called `"MyColumn"` and `"mycolumn"`. These column names would be identical using `casefold`.
Different databases may use different quoting characters for column names, eg ", ', \`. Also, in pandas column names are case sensitive and spark can be configurable to have case sensitive column names.

To fix this, making a better error message might be easier than changing the behavior of the expectation.